README.md

```diff
@@ -1 +1,65 @@
-# Namespace Node Affinity
+## Namespace Node Affinity Operator
+
+This Charm deploys a modified version of the [Namespace Node Affinity](https://github.com/idgenchev/namespace-node-affinity) Kubernetes MutatingWebhook. 
+
+The Namespace Node Affinity webhook allows a user to add a given set of node affinities and/or tolerations to all pods deployed in a namespace.  This is useful for example in a case where you have a cluster that has some nodes with specific labels (eg: nodes labeled `control-plane`) and you want all workloads in a Kubernetes namespace to be deployed only on those nodes and not any others in the cluster.  More descriptions of the tool are given in the [upstream README.md](https://github.com/idgenchev/namespace-node-affinity).
+
+## Usage
+
+This charm is deployed using the Juju command line tool as follows:
+
+```bash
+juju deploy namespace-node-affinity --trust
+```
+
+By default, the webhook is not configured to modify pods in any namespace.  To add namespaces to its scope, the user must:
+* provide a `settings_yaml` config file
+* label any namespace we want to work on with the label `namespace-node-affinity=enabled`
+
+These configurations can be modified during charm runtime, and the webhook always uses the most up to date value.  
+
+### Defining `settings_yaml`
+
+We must provide the `settings_yaml` config, which is a YAML string as described [upstream](https://github.com/idgenchev/namespace-node-affinity/blob/42674ec6863d38cbc1009e2f83243a5782aa608a/examples/sample_configmap.yaml#L8).  For example, we can configure the tool to apply:
+
+* apply a node affinity for pods in `testing-ns-a` to look for pods with the label `control-plane=true`, but only to pods that do not have the label `ignoreme: ignored`
+* apply a node affinity for pods in `testing-ns-b` to look for pods with the label `other-key: other-value`
+
+by setting the charm config:
+
+```bash
+cat <<EOF > settings.yaml
+
+testing-ns-a: |
+  nodeSelectorTerms:
+    - matchExpressions:
+      - key: control-plane
+        operator: In
+        values:
+        - true
+  excludedLabels:
+    ignoreme: ignored
+testing-ns-b: |
+  nodeSelectorTerms:
+    - matchExpressions:
+      - key: other-key
+        operator: In
+        values:
+        - other-value
+EOF
+SETTINGS_YAML=$(cat settings.yaml)
+juju config namespace-node-affinity settings_yaml="$SETTINGS_YAML"
+```
```

Comment:
I know this has already merged, I'd parked this until tomorrow and am now a bit late, but...

This is not an experience I would like to push, even though I acknowledge that the very existence of this operator is indeed a workaround. My understanding is that we need a very small part of this tool's functionality, and thus should be able to design a much more simple config schema, such as a list of namespaces (for example).

Broad application of unverified, unchecked config like this has the potential for much broader consequences, and I think we need to tighten it up. Juju is not `kubectl`, and we should avoid designing charms as such wherever possible.

Please consider alternative methods of configuring this operator that can keep the burden on human operators down, and the chance of intended outcome high :slightly_smiling_face: 

@DomFleischmann @ca-scribner